src/app/clusters/ota-requestor/OTARequestor.h

```diff
@@ -39,6 +41,11 @@ class OTARequestor : public OTARequestorInterface
     // A setter for the delegate class pointer
     void SetOtaRequestorDriver(OTARequestorDriver * driver) { mOtaRequestorDriver = driver; }
 
+    // TODO: this should really be OTADownloader, but right now OTARequestor has information that we need to initialize a
+    // BDXDownloader specifically.
+    // The BDXDownloader instance should already have the ImageProcessingDelegate set.
+    void SetBDXDownloader(chip::BDXDownloader * downloader) { mBdxDownloader = downloader; }
```

Comment:
We should be able to do this right? We're already doing something similar for setting ImageProcessor the same way in OTADownloader.h (instantiate a Linux image processor and pass it into `SetImageProcessorDelegate` which stores a pointer to `OTAImageProcessorInterface`.

```suggestion
    void SetDownloader(OTADownloader * downloader) { mOTADownloader = downloader; }
```

The caller can instantiate a new BDXDownloader and pass that into `SetDownloader`?

Reply:
The way I've written the `BDXDownloader`, it requires some extra calls for initialization that shouldn't be part of `OTADownloader` (for example, hooking it up to an `ExchangeContext`). I think that logic could be moved out of `OTARequestor.cpp` right now, I just didn't have time to do it in this PR

Reply:
Ah yes, ok nevermind.